go/proof.go

```diff
@@ -277,30 +292,46 @@ func (p *NonExistenceProof) Verify(spec *ProofSpec, root CommitmentRoot, key []b
 }
 
 // IsLeftMost returns true if this is the left-most path in the tree, excluding placeholder (empty child) nodes
-func IsLeftMost(spec *InnerSpec, path []*InnerOp) bool {
-	minPrefix, maxPrefix, suffix := getPadding(spec, 0)
+func IsLeftMost(spec *InnerSpec, path []*InnerOp) (bool, error) {
+	minPrefix, maxPrefix, suffix, err := getPadding(spec, 0)
+	if err != nil {
+		return false, err
+	}
 
 	// ensure every step has a prefix and suffix defined to be leftmost, unless it is a placeholder node
 	for _, step := range path {
-		if !hasPadding(step, minPrefix, maxPrefix, suffix) && !leftBranchesAreEmpty(spec, step) {
-			return false
+		leftBranchesAreEmpty, err := leftBranchesAreEmpty(spec, step)
+		if err != nil {
+			return false, err
+		}
+
+		if !hasPadding(step, minPrefix, maxPrefix, suffix) && !leftBranchesAreEmpty {
+			return false, nil
```

Comment:
todo: did this short circuit before if `hasPadding` returned true? (ie would that have an effect if we now call left branches first?

Reply:
I decided to change the code such that it would execute with the exact same code trace as before:
@crodriguezvega changed it to:
```go
leftBranchesAreEmpty, err := leftBranchesAreEmpty(spec, step)
if err != nil {
	return false, err
}

if !hasPadding(step, minPrefix, maxPrefix, suffix) && !leftBranchesAreEmpty {
	return false, nil
}
```

I changed it to:
```go
if !hasPadding(step, minPrefix, maxPrefix, suffix) {
        leftBranchesAreEmpty, err := leftBranchesAreEmpty(spec, step)
        if err != nil {
            return false, err 
        }   

        if !leftBranchesAreEmpty {
            return false, nil 
        }   
}
```

Because golang has short circuiting, `!hasPadding` would have previously resulted in `rightBranchesAreEmpty()` not being called, but in the code @crodriguezvega refactored, it would have been called on every execution. I tried to look at the correctness of such change, if `rightBranchAreEmpty` would function regardless of the determined padding, but it was too exhausting of a mental exercise to justify the minor differences in readability, so I opt'd to leave execution as it was before

cc @AdityaSripal